test/NuGet.Clients.Tests/NuGet.CommandLine.Test/FileSystemBackedV3MockServer.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+
+namespace NuGet.CommandLine.Test
+{
+    public class FileSystemBackedV3MockServer : MockServer
+    {
+        private string _packageDirectory;
+        private readonly MockResponseBuilder _builder;
+        public FileSystemBackedV3MockServer(string packageDirectory)
+        {
+            _packageDirectory = packageDirectory;
+            _builder = new MockResponseBuilder(Uri.TrimEnd(new[] { '/' }));
+            InitializeServer();
+        }
+
+        public ISet<PackageIdentity> UnlistedPackages { get; } = new HashSet<PackageIdentity>();
+
+        public string ServiceIndexUri => Uri + _builder.GetV3IndexPath();
+
+        private void InitializeServer()
+        {
+            Get.Add(
+                _builder.GetV3IndexPath(),
+                request =>
+                {
+                    return new Action<HttpListenerResponse>(response =>
+                    {
+                        var mockResponse = _builder.BuildV3IndexResponse(this);
+                        response.ContentType = mockResponse.ContentType;
+                        SetResponseContent(response, mockResponse.Content);
+                    });
+                });
+
+            Get.Add("/", request =>
+            {
+                return ServerHandlerV3(request);
+            });
+
+        }
+
+        private Action<HttpListenerResponse> ServerHandlerV3(HttpListenerRequest request)
+        {
+            try
+            {
+                var path = GetRequestUrlAbsolutePath(request);
+                var parts = request.Url.AbsolutePath.Split(new[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
+
+                if (path.StartsWith("/flat/") && path.EndsWith("/index.json"))
+                {
+                    return new Action<HttpListenerResponse>(response =>
+                    {
+                        response.ContentType = "text/javascript";
```

Comment:
This MIME type is obsolete. Should this be `application/javascript` instead?

Reply:
I can change it...this is mostly boilerplate that I copied around and stitched together.

Reply:
nevermind, then!

Reply:
I have fixed it.